src/main/java/org/embulk/input/sftp/SftpFileInput.java

```diff
@@ -177,7 +180,13 @@ public static String getRelativePath(PluginTask task, Optional<String> uri)
                 String uriString = uri.get();
                 String scheme = UriParser.extractScheme(uriString);
                 if (StringUtils.isEmpty(scheme)) {
-                    return GenericFileNameParser.getInstance().parseUri(null, null, uriString).getPath();
+                    try {
+                        return GenericFileNameParser.getInstance().parseUri(null, null, uriString).getPath();
+                    }
+                    catch (IllegalArgumentException ignoreEx) {
+                        //backward compatible, old version of GenericFileNameParser didn't strictly verify the uri
+                        return uriString;
+                    }
```

Comment:
Can you elaborate more on where that exception is thrown? I went through here: https://github.com/apache/commons-vfs/blob/4a488cbd2770dfec8a12a0a406822e5a9b6ed953/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/local/LocalFileNameParser.java#L62-L89 and did not see any exception.

Reply:
it will throw exception when`URI.create(uri)` on line https://github.com/apache/commons-vfs/blob/4a488cbd2770dfec8a12a0a406822e5a9b6ed953/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/local/LocalFileNameParser.java#L93, for old version `uriString` is could be invalid URI

Reply:
@vietnguyen-td , thanks for explanation. There are 2 things in my mind:
* I acknowledged that even in the old implementation, it does some cleanup, transformation too: https://github.com/apache/commons-vfs/blob/ed385689d991caf93fc498c209515f4d9db36bf4/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/local/LocalFileNameParser.java#L61-L82. If returns `uriString` itself, I'm afraid it would lead to unexpected state, thus I prefer exception is thrown instead of having the broken data.
* The URI there is actually `last path` which should not invalid URI in any case: https://github.com/embulk/embulk-input-sftp/blob/0313eef27b9939302d7fdd12e314841a3160b8d2/src/main/java/org/embulk/input/sftp/SftpFileInputPlugin.java#L36.

What do you think if we leave as it is?

Reply:
yes, nice to know `last path which should not invalid URI in any case`,  removed the `catch` block